konnectivity-client/pkg/client/client_test.go

```diff
@@ -643,3 +663,21 @@ func (s *proxyServer) handleData(pkt *client.Packet) *client.Packet {
 		},
 	}
 }
+
+func assertNoClientDialFailures(t testing.TB) {
+	t.Helper()
+	if err := metricstest.ExpectClientDialFailures(nil); err != nil {
+		t.Errorf("Unexpected %s metric: %v", "dial_failure_total", err)
+	}
+}
+
+func expectCleanShutdown(t testing.TB) {
+	metrics.Metrics.RegisterMetrics(prometheus.DefaultRegisterer)
```

Comment:
I think this only works because you use a `sync.Once` in `RegisterMetrics`. It's probably OK as is, but you might also consider moving it to an init function or TestMain.

Reply:
agreed - moved to TestMain